### PR TITLE
[BCDA-4532] Add capability to ingest Insights data from BCDA dev and test

### PR DESCRIPTION
**Pull request by msnook**
### Change Details

This PR adds glue catalog tables for BCDA's lowest environments (`dev` and `test`) in order to run various test scenarios in low environments prior to promoting to `opensbx` and `prod`.

In addition, the lambda which updates the data partitions daily was modified to include these new tables.

### Acceptance Validation

<details><summary>Click to see Terraform Plan output</summary>
<p>

```terraform
Terraform will perform the following actions:

  # aws_lambda_function.bcda_load_partitions will be updated in-place
  ~ resource "aws_lambda_function" "bcda_load_partitions" {
        arn                            = "arn:aws:lambda:us-east-1:<redacted>:function:bcda_load_partitions"
        description                    = "Loads partitions for tables"
        filename                       = "./lambda.zip"
        function_name                  = "bcda_load_partitions"
        handler                        = "index.handler"
        id                             = "bcda_load_partitions"
        invoke_arn                     = "arn:aws:apigateway:us-east-1:lambda:path/2015-03-31/functions/arn:aws:lambda:us-east-1:<redacted>:function:bcda_load_partitions/invocations"
      ~ last_modified                  = "2021-02-12T16:08:11.098+0000" -> (known after apply)
        layers                         = []
        memory_size                    = 128
        publish                        = false
        qualified_arn                  = "arn:aws:lambda:us-east-1:<redacted>:function:bcda_load_partitions:$LATEST"
        reserved_concurrent_executions = -1
        role                           = "arn:aws:iam::<redacted>:role/BcdaLoadPartitions-20210212142716462600000002"
        runtime                        = "nodejs10.x"
      ~ source_code_hash               = "sUjnHEUqAppnLLL7Y08SOzfjL4xC4U4GJuX2MNQT7pc=" -> "A5yMoropUasQbaEACDX0bG6GSKwdilVzC/SVH80Xh5k="
        source_code_size               = 657
        tags                           = {}
        timeout                        = 90
        version                        = "$LATEST"

        tracing_config {
            mode = "PassThrough"
        }
    }

  # module.insights_data_table_dev.aws_glue_catalog_table.aws_glue_catalog_table will be created
  + resource "aws_glue_catalog_table" "aws_glue_catalog_table" {
      + arn           = (known after apply)
      + catalog_id    = (known after apply)
      + database_name = "bcda"
      + description   = "dev insights data"
      + id            = (known after apply)
      + name          = "dev_insights"
      + parameters    = {
          + "EXTERNAL" = "TRUE"
        }
      + table_type    = "EXTERNAL_TABLE"

      + partition_keys {
          + comment = "Approximate delivery time"
          + name    = "dt"
          + type    = "string"
        }

      + storage_descriptor {
          + input_format  = "org.apache.hadoop.mapred.TextInputFormat"
          + location      = "s3://bfd-insights-bcda-<redacted>/databases/bcda/dev_insights"
          + output_format = "org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat"

          + columns {
              + name = "name"
              + type = "string"
            }
          + columns {
              + name = "timestamp"
              + type = "bigint"
            }
          + columns {
              + name = "json_result"
              + type = "string"
            }

          + ser_de_info {
              + name                  = "dev_insights"
              + parameters            = {
                  + "dots.in.keys"          = "true"
                  + "ignore.malformed.json" = "true"
                  + "timestamp.formats"     = "yyyy-MM-dd'T'HH:mm:ss.SSSSSS,yyyy-MM-dd'T'HH:mm:ss.SSS,yyyy-MM-dd'T'HH:mm:ss,yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z',yyyy-MM-dd'T'HH:mm:ss.SSS'Z',yyyy-MM-dd'T'HH:mm:ss'Z'"
                }
              + serialization_library = "org.apache.hive.hcatalog.data.JsonSerDe"
            }
        }
    }

  # module.insights_data_table_test.aws_glue_catalog_table.aws_glue_catalog_table will be created
  + resource "aws_glue_catalog_table" "aws_glue_catalog_table" {
      + arn           = (known after apply)
      + catalog_id    = (known after apply)
      + database_name = "bcda"
      + description   = "test insights data"
      + id            = (known after apply)
      + name          = "test_insights"
      + parameters    = {
          + "EXTERNAL" = "TRUE"
        }
      + table_type    = "EXTERNAL_TABLE"

      + partition_keys {
          + comment = "Approximate delivery time"
          + name    = "dt"
          + type    = "string"
        }

      + storage_descriptor {
          + input_format  = "org.apache.hadoop.mapred.TextInputFormat"
          + location      = "s3://bfd-insights-bcda-<redacted>/databases/bcda/test_insights"
          + output_format = "org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat"

          + columns {
              + name = "name"
              + type = "string"
            }
          + columns {
              + name = "timestamp"
              + type = "bigint"
            }
          + columns {
              + name = "json_result"
              + type = "string"
            }

          + ser_de_info {
              + name                  = "test_insights"
              + parameters            = {
                  + "dots.in.keys"          = "true"
                  + "ignore.malformed.json" = "true"
                  + "timestamp.formats"     = "yyyy-MM-dd'T'HH:mm:ss.SSSSSS,yyyy-MM-dd'T'HH:mm:ss.SSS,yyyy-MM-dd'T'HH:mm:ss,yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z',yyyy-MM-dd'T'HH:mm:ss.SSS'Z',yyyy-MM-dd'T'HH:mm:ss'Z'"
                }
              + serialization_library = "org.apache.hive.hcatalog.data.JsonSerDe"
            }
        }
    }

Plan: 2 to add, 1 to change, 0 to destroy.

------------------------------------------------------------------------
```

</p>
</details>

### Feedback Requested

All feedback is welcome.

### External References

- [BCDA-4532](https://jira.cms.gov/browse/BCDA-4532)

### Security Implications

- [ ] new software dependencies
- [ ] altered security controls
- [ ] new data stored or transmitted
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications
- [x] PHI/PII is affected by this change

### CODE REVIEW

**Review by keithdadkins — DISMISSED** [review]
LGTM

File: insights/terraform/projects/bcda/lambda/index.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ const execute_query = (env) => {
 }
 
 const handler = (event, context, callback) => {
-      Promise.all(['prod','opensbx'].map(x => execute_query(x)))
+      Promise.all(['prod','opensbx','test','dev'].map(x => execute_query(x)))
```
Comment (orenfromberg) [inline insights/terraform/projects/bcda/lambda/index.js:37]: I see a small opportunity to make this DRY:
```node
const envs = ['prod','opensbx','test','dev'];
...
if (!envs.includes(env)) {
...
Promise.all(envs.map(x => execute_query(x)))
...
```

File: insights/terraform/projects/bcda/main.tf [hunk 2]
```diff
@@ -60,6 +60,38 @@ module "glue_jobs" {
...
+module "insights_data_table_dev" {
+  source      = "../../modules/table"
+  database    = module.database.name
+  table       = "dev_insights"
+  description = "dev insights data"
+  bucket      = module.bucket.id
+  bucket_cmk  = module.bucket.bucket_cmk
+  tags        = local.tags
+  partitions  = local.partitions
+  columns = [
```
Comment (orenfromberg) [inline insights/terraform/projects/bcda/main.tf:72]: can the columns be made into a local and used in these tables to make the code more DRY?

**Review by orenfromberg — CHANGES_REQUESTED** [review]
Two small suggestions to make the code DRYer but otherwise looks good to me.

**Comment by msnook** [comment]
> Two small suggestions to make the code DRYer but otherwise looks good to me.

Great suggestions!  Thanks!  I've updated as part of the latest commit.

**Review by orenfromberg — APPROVED** [review]
:shipit:

**Review by keithdadkins — APPROVED** [review]
Looks good

**Pull request merged**